homeassistant/components/melcloud/sensor.py

```diff
@@ -207,9 +203,20 @@ def __init__(
             description.key = f"{description.key}-zone-{zone.zone_index}"
         super().__init__(api, description)
         self._zone = zone
-        self._attr_name = f"{api.name} {zone.name} {description.name}"
 
     @property
-    def native_value(self):
+    def device_info(self) -> DeviceInfo:
```

Comment:
This looks good! You can move this to a shorthand attribute `_attr_device_info`.